src/user-event/utils/dispatch-event.ts

```diff
@@ -26,12 +50,28 @@ export function dispatchHostEvent(
   });
 }
 
+function findEnabledEventHandler(
+  element: ReactTestInstance,
+  eventName: string
+): EventHandler | null {
+  const handler = getEnabledEventHandler(element, eventName);
+  if (handler) {
+    return handler;
+  }
+
+  const parent = getHostParent(element);
+  if (!parent) {
+    return null;
+  }
+
+  return findEnabledEventHandler(parent, eventName);
+}
+
 function getEnabledEventHandler(
   element: ReactTestInstance,
   eventName: string
 ): EventHandler | null {
   const touchResponder = isTouchResponder(element) ? element : undefined;
-
   const handler = getEventHandler(element, eventName);
   if (handler && isEventEnabled(element, eventName, touchResponder)) {
```

Comment:
I don't think we shoud use the function isEventEnabled for userEvent. It is used for any kind of events and supports every eventName, whereas here we could have a pretty straightforward check on the editable prop. We know we want to type something and not do something else and that allows better handling for specific events, which fireEvent lacks because it's generic and I think we should use that for userEvent

Reply:
I have to check experimentally if parent `pointerEvents` could block `TextInput` focus, if not then we could just inspect the `editable` prop.

Reply:
Oh I hadn't thought about that, I'll probably need to check for pointer events as well for Text and TextInput for user.press

Reply:
So on iOS: pointerEvents="none" on either View wrapping TextInput or TextInput itself prevents TextInput from gaining focus
On Android:  pointerEvents="none" on either View wrapping TextInput or prevents TextInput from gaining focus, but on TextInput itself is not preventing focus 🤦‍♂️